src/components/Modal/VotingPowerDelegationDetail/VotingPowerDelegationDetail.tsx

```diff
@@ -12,14 +31,96 @@ type VotingPowerDelegationDetailProps = {
 
 function VotingPowerDelegationDetail({ delegate, onBackClick }: VotingPowerDelegationDetailProps) {
   const t = useFormatMessage()
+  const { address } = delegate
+  const { votingPower } = useVotingPowerBalance(address)
```

Comment:
This reminded me of the `isLoadingVotingPower`... 
Maybe we could have a loader show if any of this information is missing?
(I think most of these methods already come with a `state` we can use)

Reply:
Right! Let's handle all the loading cases after joining all the pieces of this modal